scanner/services/converters/converters.go

```diff
@@ -0,0 +1,400 @@
+package converters
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/pkg/cpe"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+)
+
+var (
+	severityMapping = map[claircore.Severity]v4.VulnerabilityReport_Vulnerability_Severity{
+		claircore.Unknown:    v4.VulnerabilityReport_Vulnerability_SEVERITY_UNSPECIFIED,
+		claircore.Negligible: v4.VulnerabilityReport_Vulnerability_SEVERITY_LOW,
+		claircore.Low:        v4.VulnerabilityReport_Vulnerability_SEVERITY_LOW,
+		claircore.Medium:     v4.VulnerabilityReport_Vulnerability_SEVERITY_MODERATE,
+		claircore.High:       v4.VulnerabilityReport_Vulnerability_SEVERITY_IMPORTANT,
+		claircore.Critical:   v4.VulnerabilityReport_Vulnerability_SEVERITY_CRITICAL,
+	}
+)
+
+// ToProtoV4IndexReport maps claircore.IndexReport to v4.IndexReport.
+func ToProtoV4IndexReport(r *claircore.IndexReport) *v4.IndexReport {
+	if r == nil {
+		return nil
+	}
+	return &v4.IndexReport{
+		State:    r.State,
+		Success:  r.Success,
+		Err:      r.Err,
+		Contents: toProtoV4Contents(r.Packages, r.Distributions, r.Repositories, r.Environments),
```

Comment:
Should the `hash`/`hashId` be carried over as well?

Reply:
No, we don't use the internal HashID from ClairCore. Instead, we use our own HashID that includes the resource type as a namespace. We plan to scan things besides container images, like nodes.